### DISCUSSION TITLE
Question about MOOSE output

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#22086] [Q&A Modules: Phase field] [answered]
Hi everyone,

I am running simulations with the CH equation using MOOSE. I am using the split form for solving this CH equation in MOOSE.
The following is the CH equation and chemical potential.
![image](https://user-images.githubusercontent.com/63679626/190019134-71cdb981-0988-4adf-b635-597866cf40ad.png)
![image](https://user-images.githubusercontent.com/63679626/190019162-a7b53d39-eece-4fbb-b805-a7f2052e7701.png)
This is the kernels I used to solve the CH equation.
![image](https://user-images.githubusercontent.com/63679626/190019359-96a3d238-ff11-462b-a531-49b831f29073.png)
The following is the flux. I want to plot the two red parts shown in the flux's expression. One is the derivative of free energy function wrt concentration as a function of location, another is Laplacian of concentration as a function of location.
How can I directly get these two parts' value from my MOOSE output file? What are the names of these two parts in the output file?
![image](https://user-images.githubusercontent.com/63679626/190019214-65713f4a-f7c5-42ce-afc4-ebc7d85dd42e.png)
Could you please give me some advice?

Thank you very much.
Best,

**Comment by GiudGiud** [comment, marked as answer]
@dschwen does this already exist as an auxiliary kernel?

If not, you ll have to make an auxkernel to output this

**Reply by abc-hy** [reply]
Is there a aux kernel for this? @GiudGiud 
How to make an auxkernel?

**Reply by GiudGiud** [reply]
I doubt it. Daniel will confirm.

You should go through the application dev tutorial, or the main tutorial to learn about this
https://mooseframework.inl.gov/workshop/index.html#/
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

**Reply by abc-hy** [reply]
There should be direct way to output this....
Do you know how to output derivative of free energy function wrt c?

**Reply by abc-hy** [reply]
I searched DerivativeParsedMaterial in the MOOSE website, and Currently I added the material_property_names='dF:=D[F,c]' in the free energy function block to calculate the derivative of free energy function wrt concentration. I am not sure. Is this correct? Will this influence my free energy function? Could you please give me some advice?
![image (1)](https://user-images.githubusercontent.com/63679626/190285167-1f6a562d-9c79-4bae-ba9d-1de31bbb54bc.png)

**Reply by GiudGiud** [reply]
Addind the derivative will not influence you free energy function
I m not sure the derivative gets output to exodus. Do you see it?
It gets defined for sure so you could use a MaterialRealAux to output it
